src/bitbox02.js

```diff
@@ -99,87 +113,72 @@ export class BitBox02API {
      * @param setStatusCb Callback that lets the API set the status received from the device.
      * @return Promise that will resolve once the pairing is complete.
      */
-    connect (showPairingCb, userVerify, handleAttestationCb, onCloseCb, setStatusCb) {
-        const self = this;
-        self.opened = true;
-        return new Promise((resolve, reject) => {
-            function onWrite(bytes) {
-                if (self.socket.readyState != WebSocket.OPEN) {
-                    console.log("Error, trying to write to closed socket");
-                    return;
-                }
-                self.socket.send(bytes);
+    async connect(showPairingCb, userVerify, handleAttestationCb, onCloseCb, setStatusCb) {
+        this.opened = true;
+        this.socket = await this.connectWebsocket();
+        this.socket.onclose = function(event) {
+                onCloseCb();
+            }
+        this.socket.onmessage = event => {
+            this.firmware().js.OnRead(new Uint8Array(event.data));
+        }
+        const onWrite = bytes => {
+            if (this.socket.readyState != WebSocket.OPEN) {
+                console.log("Error, trying to write to closed socket");
```

Comment:
Use single quotes

Reply:
out of scope, i just moved existing code around - there are lots of other double quotes too.

would be great if we could add some linting so this is part of the process, not part of reviews.